main/src/library/LazyList.flix

```diff
@@ -14,57 +14,47 @@
  * limitations under the License.
  */
 
+pub enum LazyList[a] {
+    case ENil,
+    case ECons(a, LazyList[a]),
+    case LCons(a, Lazy[LazyList[a]]),
+    case LList(Lazy[LazyList[a]])
+}
+
 namespace LazyList {
 
     ///
-    /// An immutable lazy list.
-    ///
-    /// A lazy list is either the empty list represented by `Empty`, or
-    /// an element `x` followed by a lazy list `xs`.
-    /// Note that a lazy list is lazy on its tail but strict on its head.
-    /// The empty list is always strict.
+    /// Returns an empty LazyList.
     ///
-    pub enum LazyList[t] {
-        case Empty,
-        case LazyCons(t, Lazy[LazyList[t]])
-    }
+    pub def empty(): LazyList[a] = LList(lazy ENil)
 
     ///
-    /// Returns the empty LazyList.
+    /// Alias for `empty`.
+    ///
+    /// Returns an empty LazyList.
     ///
-    pub def empty(): LazyList[a] = Empty
+    pub def new(): LazyList[a] = empty()
```

Comment:
Do we need both `new` and `empty`?

Reply:
I think one should be sufficient, and I think it should be called `empty` in this case, because LazyList is an immutable kind of thing.